src/java.base/share/classes/java/lang/StringTemplate.java

```diff
@@ -0,0 +1,692 @@
+     *
+     * @return combined {@link StringTemplate}
+     *
+     * @throws NullPointerException if stringTemplates is null or if any of the
+     * its elements are null
+     *
+     * @implNote If {@code stringTemplates.size() == 0} then a {@link StringTemplate} with
+     * an empty fragment and no values is returned, as if invoking
+     * <code>StringTemplate.of("")</code> . If {@code stringTemplates.size() == 1}
+     * then the first element of the list is returned unchanged.
+     */
+    static StringTemplate combine(List<StringTemplate> stringTemplates) {
+        JavaTemplateAccess JTA = SharedSecrets.getJavaTemplateAccess();
+        return JTA.combine(stringTemplates.toArray(new StringTemplate[0]));
+    }
+
+    /**
+     * This {@link StringProcessor} instance is conventionally used for the string interpolation
+     * of a supplied {@link StringTemplate}.
+     * <p>
+     * For better visibility and when practical, it is recommended that users use the
+     * {@link StringTemplate#STR} processor instead of invoking the
+     * {@link StringTemplate#interpolate()} method.
+     * Example: {@snippet :
+     * int x = 10;
+     * int y = 20;
+     * String result = STR."\{x} + \{y} = \{x + y}"; // @highlight substring="STR"
+     * }
+     * In the above example, the value of {@code result} will be {@code "10 + 20 = 30"}. This is
+     * produced by the interleaving concatenation of fragments and values from the supplied
+     * {@link StringTemplate}. To accommodate concatenation, values are converted to strings
+     * as if invoking {@link String#valueOf(Object)}.
+     * @implNote {@link StringTemplate#STR} is statically imported implicitly into every
+     * Java compilation unit.<p>The result of interpolation is not interned.
+     */
+    static final StringProcessor STR = StringTemplate::interpolate;
+
+    /**
+     * This {@link SimpleProcessor} instance is conventionally used to indicate that the
+     * processing of the {@link StringTemplate} is to be deferred to a later time. Deferred
+     * processing can be resumed by invoking the
+     * {@link StringTemplate#process(Processor)} or
+     * {@link Processor#process(StringTemplate)} methods.
+     * {@snippet :
+     * import static java.lang.StringTemplate.RAW;
+     * ...
+     * StringTemplate st = RAW."\{x} + \{y} = \{x + y}";
+     * ...other steps...
+     * String result = STR.process(st);
+     * }
+     * @implNote Unlike {@link StringTemplate#STR}, {@link StringTemplate#RAW} must be
+     * statically imported explicitly.
+     */
+    static final SimpleProcessor<StringTemplate> RAW = st -> st;
+
+    /**
+     * This interface describes the methods provided by a generalized string template processor. The
+     * primary method {@link Processor#process(StringTemplate)} is used to validate
+     * and compose a result using a {@link StringTemplate StringTemplate's} fragments and values lists.
```

Comment:
nit: `{@link StringTemplate StringTemplate's}` will probably not render as you'd expect (as it will all be preformat, including the `'s`).

Reply:
works for me